src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const partsOfDate = date.split(oldSeparator);
+  const dateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObject[fromFormat[i]] = partsOfDate[i];
+  }
+
+  const newSeparator = toFormat[3];
+  const newPartsOfDate = toFormat.slice(0, 3).map((part) => {
+    if (part === 'YY' && dateObject['YYYY']) {
+      return dateObject['YYYY'].slice(2);
+    } else if (part === 'YYYY' && dateObject['YY']) {
+      const year = parseInt(dateObject['YY'], 10);
+
+      return year < 30 ? `20${dateObject['YY']}` : `19${dateObject['YY']}`;
+    }
+
+    return dateObject[part];
+  });
+
+  return newPartsOfDate.join(newSeparator);
```

Comment:
The `formatDate` function correctly implements the required functionality of reformatting the date according to the given formats. However, there are a couple of suggestions to improve the code:
- Consider adding error handling for unexpected cases, even though the input is assumed to be correct. This can help in maintaining the code in the future if the assumptions change.
- The use of `parseInt` in line 24 is good for converting strings to numbers, but since we're not performing any calculations, it might be more straightforward to use string concatenation for the year conversion in lines 26-27.
- To improve readability, you may consider creating a separate function for the conversion of the year format. This would make the main `formatDate` function cleaner and more focused on its primary task.